README.md

```diff
@@ -1,9 +1,205 @@
+that do not have high-performance fabric hardware.  The sockets provider
+supports all libfabric provider requirements and interfaces.
+
+See the `fi_sockets(7)` man page for more details.
+
+### usNIC
+
+***
+
+The `usNIC` provider is designed to run over the Cisco VIC (virtualized NIC)
+hardware on Cisco UCS servers. It utilizes the Cisco usNIC (userspace NIC)
+capabilities of the VIC to enable ultra low latency and other offload
+capabilites on Ethernet networks.
+
+See the `fi_usnic(7)` man page for more details.
+
+#### Dependencies
+
+- The `usNIC` provider depends on having either `libnl1` or `libnl3`. If the
+  provider is being built from source, then the `libnl` header files will need
+  to be provided (e.g. install the "--devel" versions of the package).
+
+#### Configure options
+
+```
+--with-libnl=<directory>
+```
+
+If specified, look for libnl support. If it is not found then the `usNIC`
+provider will not be built. If `<directory>` is specified, then check in the
+directory and check for `libnl3`. If `libnl3` is not found, then check for
+`libnl1`. If no `<directory>` argument is specified, then this option is
+redundant with `--with-usnic` (default).
+
+### GNI
+
+***
+
+The `GNI` provider runs on Cray XC (TM) systems utilizing the user-space
+Generic Network Interface (`uGNI`) which provides low-level access to
+the Aries interconnect.  The Aries interconnect is designed for
+low-latency one-sided messaging and also includes direct hardware
+support for common atomic operations and optimized collectives.
+
+See the `fi_gni(7)` man page for more details.
+
+#### Dependencies
+
+- The `GNI` provider requires `gcc` version 4.9 or higher.
+
+### MXM
+
+***
+
+The `mxm` provider runs over the MXM (Mellanox messaging) interface that is
+currently supported by the Mellanox infiniband fabrics.  The `mxm` provider
+makes use of MXM tag matching API in order to implement a limited set of the
+libfabric data transfer APIs, namely, tagged message queue.
+
+See the `fi_mxm(7)` man page for more details.
+
+
+### UDP
+
+***
+
+The `UDP` provider is a basic provider that can be used on any system that
+supports UDP sockets.  The provider is not intended to provide performance
+improvements over regular TCP sockets, but rather to allow application and
+provider developers to write, test, and debug their code.  The `UDP` provider
+forms the foundation of a utility provider that enables the implementation of
+libfabric features over any hardware.
+
+See the `fi_udp(7)` man page for more details.
+
+
+### PSM
+
+***
+
+The `psm` provider runs over the PSM 1.x interface that is currently supported
+by the Intel TrueScale Fabric. PSM provides tag-matching message queue
+functions that are optimized for MPI implementations.  PSM also has limited
+Active Message support, which is not officially published but is quite stable
+and well documented in the source code (part of the OFED release). The `psm`
```

Comment:
I took this straight from the intro paragraph in `fi_psm(7)`. So I don't know...